libs/cardano-ledger-test/src/Test/Cardano/Ledger/Constrained/Lenses.hs

```diff
@@ -183,3 +202,29 @@ strictSeqListL = lens toList (\_ y -> fromList y)
 
 mapCompactFormCoinL :: Lens' (Map a (CompactForm Coin)) (Map a Coin)
 mapCompactFormCoinL = lens (Map.map fromCompact) (\_ y -> Map.map compactCoinOrError y)
+
+drepDelegationUMapL :: Lens' (UMap c) (Map (Credential 'Staking c) (DRep c))
+drepDelegationUMapL = lens dRepMap delta
+  where
+    delta um new = unSplitUMap (split {spDRep = new})
+      where
+        split = splitUMap um
+
+pairL :: Lens' x y -> Lens' a b -> Lens' (x, a) (y, b)
+pairL xy ab = lens getter setter
+  where
+    getter (x, a) = (x ^. xy, a ^. ab)
+    setter (x, a) (y, b) = (x & xy .~ y, a & ab .~ b)
+
+firstL :: Lens' (a, b) a
+firstL = lens fst (\(_, y) a -> (a, y))
+
+secondL :: Lens' (a, b) b
+secondL = lens snd (\(x, _) b -> (x, b))
```

Comment:
There are already lenses called `_1` and `_2` that do the same thing

Reply:
I removed them